addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.fields import Command
+from odoo.osv import expression
+from odoo.tools import html2plaintext, email_normalize
+
+from collections import defaultdict
+from markupsafe import Markup
+
+
+class ChatbotScriptStep(models.Model):
+    _name = 'chatbot.script.step'
+    _description = 'Chatbot Script Step'
+    _order = 'sequence, id'
+    _rec_name = 'message'
+
+    message = fields.Text(string='Message', translate=True)
+    sequence = fields.Integer(string='Sequence')
+    chatbot_script_id = fields.Many2one(
+        'chatbot.script', string='Chatbot', required=True, ondelete='cascade')
+    step_type = fields.Selection([
+        ('text', 'Text'),
+        ('question_selection', 'Question'),
+        ('question_email', 'Email'),
+        ('question_phone', 'Phone'),
+        ('forward_operator', 'Forward to Operator'),
+        ('free_input_single', 'Free Input'),
+        ('free_input_multi', 'Free Input (Multi-Line)'),
+    ], default='text', required=True)
+    # answers
+    answer_ids = fields.One2many(
+        'chatbot.script.answer', 'script_step_id',
+        compute='_compute_answer_ids', store=True, readonly=False,
+        copy=True, string='Answers')
+    triggering_answer_ids = fields.Many2many(
+        'chatbot.script.answer', domain="[('script_step_id.sequence', '<', sequence)]",
+        compute='_compute_triggering_answer_ids', readonly=False, store=True,
+        copy=False,  # copied manually, see chatbot.script#copy
+        string='Only If', help='Show this step only if all of these answers have been selected.')
+    # forward-operator specifics
+    is_forward_operator_child = fields.Boolean(compute='_compute_is_forward_operator_child')
+
+    @api.depends('step_type')
+    def _compute_answer_ids(self):
+        for step in self:
+            if step.step_type != 'question_selection' or not step.answer_ids:
+                step.answer_ids = False
+
+    @api.depends('sequence')
+    def _compute_triggering_answer_ids(self):
+        for step in self:
```

Comment:
``for step in self.filtered('triggering_answer_ids')`` to gain an indentation level ?